crates/materializer-general/src/lib.rs

```diff
@@ -152,18 +121,16 @@ impl GeneralMaterializer for MaterializerImpl {
             .parse()
             .map_err(anyhow::Error::from)
             .map_err(error_handler)?;
-        let view_definition = self
-            .get_view_definition(view_id)
-            .await
-            .map_err(error_handler)?;
+        let mut view_cache = self.view_cache.lock().await;
+        let view_definition = view_cache.get(view_id).await.map_err(error_handler)?;
```

Comment:
```suggestion
        {
            let mut view_cache = self.view_cache.lock().await;
            let view_definition = view_cache.get(view_id).await.map_err(error_handler)?;
        }
```
We don't have to keep mutex locked the whole time + same issue as with the previous comment. However, that might not be a problem with materialization yet - OB is still limited to process one view at a time(for now).